acceptancetests/deploy_stack.py

```diff
@@ -414,18 +414,19 @@ def copy_remote_logs(remote, directory):
             logging.warning(repr(e))
 
 
-def assess_juju_run(client):
-    responses = client.run(('uname',),
+# NOTE: Since juju 3, "juju run" has been renamed to "juju exec"
+def assess_juju_exec(client):
+    responses = client.exec_cmds(('uname',),
                            applications=['dummy-source', 'dummy-sink'])
-    for machine in responses:
-        if machine.get('ReturnCode', 0) != 0:
-            raise ValueError('juju run on machine %s returned %d: %s' % (
-                machine.get('MachineId'),
-                machine.get('ReturnCode'),
-                machine.get('Stderr')))
+    for response in responses.values():
+        if response.get('results').get('return-code', 0) != 0:
```

Comment:
It would probably be more Pythonic to say:

```
if machine.get('ReturnCode'):
    ...
```

Both the default return of .get, which is None, and 0 are Falsey values, while -1 and any number great than zero are Truthy values. You can thus just do a True/False check and be done with it.

Reply:
oh ok. let me patch that.